tox.ini

```diff
@@ -32,15 +32,11 @@ commands =
     black .
     isort .
 
-[testenv:tics]
-deps = .[unittests]
-description = Run all tests for snap and produce coverage report
-commands =
-    pytest {toxinidir}/tests \
-    -v --cov --cov-report=xml:{toxinidir}/tests/report/coverage.xml --junit-xml={toxinidir}/tests/report/tests.xml
-
 [testenv:unit]
 deps = .[unittests]
+setenv =
+    {[testenv]setenv}
+    COVERAGE_FILE = coverage-unit
```

Comment:
I would much prefer to see the coverage db file location left as the default (`.coverage`), to avoid needing these extra envvars, gitignore updates, etc. We can use renaming and such as needed when using the github upload/download actions to avoid file name conflicts where needed (we'll need something anyway, because even with the naming split between coverage-unit and func, there will be multiple coverage-unit or coverage-func files - one for each architecture tested).

Reply:
IMO this is personal taste. The changes are minimal:
- one extra env var for each test
- one extra gitignore

The benefit of the current approach is that it's easier to replicate locally.  If for some reason starts failing you can run locally, separate files will be generated to analyse. Otherwise you can easily overwrite and it's not very intuitive

Reply:
Hmm I see your point. I'm not sure I'm convinced though - the html/xml files output will still overwrite the previous one.

If we go this way, please keep them as hidden files, following the pattern of the original file (`.coverage` -> `.coverage-unit`, `.coverage-func`). That should help avoid confusion about these new files, and you can edit the existing gitignore line (`.coverage` -> `.coverage*`).

Reply:
Hmm I see your point. I'm not sure I'm convinced though - the html/xml files output will still overwrite the previous one.

If we go this way, please keep them as hidden files, following the pattern of the original file (`.coverage` -> `.coverage-unit`, `.coverage-func`). That should help avoid confusion about these new files, and you can edit the existing gitignore line (`.coverage` -> `.coverage*`).

Reply:
Ok. Changed to be hidden files